CMake build doesn't work

Build log:

```
[100%] Linking CXX executable nheqminer
/usr/bin/cmake -E cmake_link_script CMakeFiles/nheqminer.dir/link.txt --verbose=1
/usr/bin/x86_64-pc-linux-gnu-g++   -march=native -O2 -pipe  -std=c++11 -march=native   -Wl,-O1 -Wl,--as-needed CMakeFiles/nheqminer.dir/main.cpp.o CMakeFiles/nheqminer.dir/amount.cpp.o CMakeFiles/nheqminer.dir/api.cpp.o CMakeFiles/nheqminer.dir/arith_uint256.cpp.o CMakeFiles/nheqminer.dir/crypto/ripemd160.cpp.o CMakeFiles/nheqminer.dir/crypto/sha256.cpp.o CMakeFiles/nheqminer.dir/json/json_spirit_reader.cpp.o CMakeFiles/nheqminer.dir/json/json_spirit_value.cpp.o CMakeFiles/nheqminer.dir/json/json_spirit_writer.cpp.o CMakeFiles/nheqminer.dir/libstratum/StratumClient.cpp.o CMakeFiles/nheqminer.dir/libstratum/ZcashStratum.cpp.o CMakeFiles/nheqminer.dir/primitives/block.cpp.o CMakeFiles/nheqminer.dir/speed.cpp.o CMakeFiles/nheqminer.dir/trompequihash/blake2/blake2bx.cpp.o CMakeFiles/nheqminer.dir/uint256.cpp.o CMakeFiles/nheqminer.dir/utilstrencodings.cpp.o  -o nheqminer -rdynamic -lboost_system-mt -lboost_log_setup-mt -lboost_log-mt -lboost_date_time-mt -lboost_filesystem-mt -lboost_thread-mt 
/usr/lib/gcc/x86_64-pc-linux-gnu/5.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: CMakeFiles/nheqminer.dir/main.cpp.o: undefined reference to symbol 'pthread_rwlock_wrlock@@GLIBC_2.2.5'
/lib64/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/nheqminer.dir/build.make:491: nheqminer] Error 1
make[2]: Leaving directory '/var/tmp/portage/net-misc/nheqminer-9999/work/nheqminer-9999_build'
make[1]: *** [CMakeFiles/Makefile2:68: CMakeFiles/nheqminer.dir/all] Error 2
make[1]: Leaving directory '/var/tmp/portage/net-misc/nheqminer-9999/work/nheqminer-9999_build'
make: *** [Makefile:84: all] Error 2
```

Proposed patch:

```
--- nheqminer/nheqminer/CMakeLists.txt
+++ nheqminer/nheqminer/CMakeLists.txt
@@ -117,11 +117,11 @@
     )
 #add_executable(${PROJECT_NAME} ${SRC_LIST})

-set(LIBS ${LIBS} ${Threads_LIBRARIES} ${Boost_LIBRARIES})
+set(LIBS ${LIBS} ${Boost_LIBRARIES})

 #message("-- CXXFLAGS: ${CMAKE_CXX_FLAGS}")
 #message("-- LIBS: ${LIBS}")

 add_executable(${PROJECT_NAME} ${SOURCE_FILES})
 #target_link_libraries(${PROJECT_NAME} ${Boost_LIBRARIES})
-target_link_libraries(${PROJECT_NAME} ${LIBS})
+target_link_libraries(${PROJECT_NAME} ${LIBS} ${CMAKE_THREAD_LIBS_INIT})
```

See:
http://stackoverflow.com/questions/19901934/strange-linking-error-dso-missing-from-command-line
https://cmake.org/cmake/help/v2.8.8/cmake.html#module:FindThreads

--- a/Linux_cmake/nheqminer_cpu/CMakeLists.txt
+++ b/Linux_cmake/nheqminer_cpu/CMakeLists.txt
@@ -127,11 +127,11 @@ set(SOURCE_FILES
 
 #add_executable(${PROJECT_NAME} ${SRC_LIST})
 
-set(LIBS ${LIBS} ${Threads_LIBRARIES} ${Boost_LIBRARIES})
+set(LIBS ${LIBS} ${Boost_LIBRARIES})
 
 #message("-- CXXFLAGS: ${CMAKE_CXX_FLAGS}")
 #message("-- LIBS: ${LIBS}")
 
 add_executable(${PROJECT_NAME} ${SOURCE_FILES})
 #target_link_libraries(${PROJECT_NAME} ${Boost_LIBRARIES})
-target_link_libraries(${PROJECT_NAME} ${LIBS} xenoncat_avx1 xenoncat_avx2 )
+target_link_libraries(${PROJECT_NAME} ${LIBS} ${CMAKE_THREAD_LIBS_INIT} xenoncat_avx1 xenoncat_avx2 )

--- a/Linux_cmake/nheqminer_cpu_tromp/CMakeLists.txt
+++ b/Linux_cmake/nheqminer_cpu_tromp/CMakeLists.txt
@@ -133,11 +133,11 @@ set(SOURCE_FILES
 
 #add_executable(${PROJECT_NAME} ${SRC_LIST})
 
-set(LIBS ${LIBS} ${Threads_LIBRARIES} ${Boost_LIBRARIES})
+set(LIBS ${LIBS} ${Boost_LIBRARIES})
 
 #message("-- CXXFLAGS: ${CMAKE_CXX_FLAGS}")
 #message("-- LIBS: ${LIBS}")
 
 add_executable(${PROJECT_NAME} ${SOURCE_FILES})
 #target_link_libraries(${PROJECT_NAME} ${Boost_LIBRARIES})
-target_link_libraries(${PROJECT_NAME} ${LIBS})
+target_link_libraries(${PROJECT_NAME} ${LIBS} ${CMAKE_THREAD_LIBS_INIT})

--- a/Linux_cmake/nheqminer_cpu_xenoncat/CMakeLists.txt
+++ b/Linux_cmake/nheqminer_cpu_xenoncat/CMakeLists.txt
@@ -112,11 +112,11 @@ set(SOURCE_FILES
 
 #add_executable(${PROJECT_NAME} ${SRC_LIST})
 
-set(LIBS ${LIBS} ${Threads_LIBRARIES} ${Boost_LIBRARIES})
+set(LIBS ${LIBS} ${Boost_LIBRARIES})
 
 #message("-- CXXFLAGS: ${CMAKE_CXX_FLAGS}")
 #message("-- LIBS: ${LIBS}")
 
 add_executable(${PROJECT_NAME} ${SOURCE_FILES})
 #target_link_libraries(${PROJECT_NAME} ${Boost_LIBRARIES})
-target_link_libraries(${PROJECT_NAME} ${LIBS} xenoncat_avx1 xenoncat_avx2 )
+target_link_libraries(${PROJECT_NAME} ${LIBS} ${CMAKE_THREAD_LIBS_INIT} xenoncat_avx1 xenoncat_avx2 )

